harness/determined/common/experimental/determined.py

```diff
@@ -404,3 +404,157 @@ def stream_trials_validation_metrics(
             trial_ids: List of trial IDs to get metrics for.
         """
         return trial._stream_validation_metrics(self._session, trial_ids)
+
+    def get_workspace_ids(
+        self,
+        session: api.Session,
+        workspace_names: List[str],
+    ) -> List[int]:
+        wnSet = set(workspace_names)
+        workspace_ids = []
+
+        def get_with_offset(offset: int) -> bindings.v1GetWorkspacesResponse:
+            return bindings.get_GetWorkspaces(
+                session=self._session,
+                offset=offset,
+            )
+
+        resps = api.read_paginated(get_with_offset)
+        workspaces = [w for r in resps for w in r.workspaces]
+
+        for w in workspaces:
+            if w.name in wnSet:
+                if w.archived:
+                    raise Exception(f'Workspace "{w.name}" is archived.')
+                workspace_ids.append(w.id)
+        return workspace_ids
+
+    def bind_rps_to_workspaces(
+        self,
+        resource_pool_names: List[str],
+        workspace_names: List[str],
+    ) -> None:
+        """
+        Bind resource pools to workspaces.
+
+        Arguments:
+            resource_pool_names (list(str)): The names of the resouce pools to be
+                bound.
+            workspace_names (list(str)): The names of the workspaces to be bound.
+        """
+        workspace_ids = self.get_workspace_ids(self._session, workspace_names)
+
+        for resource_pool_name in resource_pool_names:
+            req = bindings.v1BindRPToWorkspaceRequest(
+                resourcePoolName=resource_pool_name,
+                workspaceIds=workspace_ids,
+            )
+
+            bindings.post_BindRPToWorkspace(
+                self._session, body=req, resourcePoolName=resource_pool_name
+            )
+
+    def unbind_rp_from_workspaces(
+        self,
+        resource_pool_name: str,
+        workspace_names: List[str],
+    ) -> None:
+        """
+        Unbind a resource pool from workspaces.
+
+        Arguments:
+            resource_pool_name (str): The name of the resouce pool to be
+                unbound.
+            workspace_names (list(str)): The names of the workspaces to be unbound.
+        """
+        workspace_ids = self.get_workspace_ids(self._session, workspace_names)
+
+        req = bindings.v1UnbindRPFromWorkspaceRequest(
+            resourcePoolName=resource_pool_name,
+            workspaceIds=workspace_ids,
+        )
+
+        bindings.delete_UnbindRPFromWorkspace(
+            self._session, body=req, resourcePoolName=resource_pool_name
+        )
+
+    def list_rps_bound_to_workspace(
```

Comment:
To fit with the rest of the SDK, I'd prefer an interface with the following translations:
* `list_rps_bound_to_workspace` -> `Workspace.resource_pools()`
* `bind_rps_to_workspaces` -> `ResourcePool.bind(workspace_ids)`
* `list_workspaces_bound_to_rp` -> `ResourcePool.workspaces()`
* `overwrite_rp_workspace_bindings` -> `ResourcePool.set_bindings(workspace_ids)`

Currently neither Workspace nor ResourcePool exist (and it's not even clear what a "ResourcePool" should represent). If you approve of this interface, I'd be happy to create them in another PR.